### PR TITLE
Update docs about Terraform provider registry

### PR DESCRIPTION
**Pull request by damianstasik**
# Description of the change

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/vendors/terraform/provider-registry.md [hunk 1]
```diff
@@ -25,7 +23,13 @@ Last but not least, we assume you're going to use GoReleaser to build your provi
 
 ### Creating a provider
 
-To create a provider in Spacelift, you will either need to use the [API](../../integrations/api.md) (`terraformProviderCreate` mutation) or - preferably - our Terraform provider. The latter is the easiest way to do it, as it will let you manage your provider declaratively in the future:
+To create a provider in Spacelift, you have three options:
+
+#### 1. Use the [API](../../integrations/api.md) (`terraformProviderCreate` mutation).
+
+#### 2. Use our Terraform provider (preferable).
```
Comment (jmfontaine) [inline docs/vendors/terraform/provider-registry.md:30]: Start with the preferred option. 😄

File: docs/vendors/terraform/provider-registry.md [hunk 2]
```diff
@@ -25,7 +23,13 @@ Last but not least, we assume you're going to use GoReleaser to build your provi
 
 ### Creating a provider
 
-To create a provider in Spacelift, you will either need to use the [API](../../integrations/api.md) (`terraformProviderCreate` mutation) or - preferably - our Terraform provider. The latter is the easiest way to do it, as it will let you manage your provider declaratively in the future:
+To create a provider in Spacelift, you have three options:
+
+#### 1. Use the [API](../../integrations/api.md) (`terraformProviderCreate` mutation).
```
Comment (jmfontaine) [inline docs/vendors/terraform/provider-registry.md:28]: Could you expand a bit on that, please? One or two paragraphs are enough to orient users.

You can probably drop the numbering.

Avoid links in titles. It looks odd.

File: docs/vendors/terraform/provider-registry.md [hunk 3]
```diff
@@ -36,16 +40,33 @@ resource "spacelift_terraform_provider" "provider" {
...
+
+### Register a GPG key
 
 !!! warning
     Only Spacelift root admins can manage account GPG keys. If you're not a root admin, you will need to ask one to do it for you.
 
-Terraform uses GPG keys to verify the authenticity of providers. Before you can publish a provider version, you need to register a GPG key with Spacelift. You can do that either using the [API](../../integrations/api.md) (`gpgKeyCreate` mutation) or - preferably - our CLI tool called [`spacectl`](https://github.com/spacelift-io/spacectl){: rel="nofollow"}. One reason we do not want to do it declaratively through the Terraform provider is that it would inevitably lead to the private key being stored in some Terraform state, which is not ideal. `spacectl` will let you register a GPG key without storing the private key anywhere outside of your system.
+Terraform uses GPG keys to verify the authenticity of providers. Before you can publish a provider version, you need to register a GPG key with Spacelift. Similarly to creating a provider, you have three options to register a GPG key:
+
+#### 1. Use the [API](../../integrations/api.md) (`gpgKeyCreate` mutation).
```
Comment (jmfontaine) [inline docs/vendors/terraform/provider-registry.md:65]: Could you expand a bit on that, please? One or two paragraphs are enough to orient users.

You can probably drop the numbering.

Avoid links in titles. It looks odd.

**Review by rjohnep — APPROVED** [review]
LGTM

**Pull request merged**